pkg/ristretto/ristretto_test.go

```diff
@@ -0,0 +1,330 @@
+package ristretto
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"encoding/hex"
+	"encoding/json"
+	"math/big"
+	"testing"
+
+	"filippo.io/edwards25519/field"
+)
+
+// The encoding of the canonical generator.
+var compressedRistrettoBasepoint, _ = hex.DecodeString("e2f2ae0a6abc4e71a884a961c500515f58e30b6aa582dd8db6a65945e08d2d76")
+
+func TestRistrettoBasepointRoundTrip(t *testing.T) {
+	decodedBasepoint := &Element{}
+	_, err := decodedBasepoint.SetCanonicalBytes(compressedRistrettoBasepoint)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ristrettoBasepoint := NewGeneratorElement()
+	if decodedBasepoint.Equal(ristrettoBasepoint) != 1 {
+		t.Error("decode succeeded, but got wrong point")
+	}
+
+	roundtripBasepoint := decodedBasepoint.Bytes()
+	if !bytes.Equal(compressedRistrettoBasepoint, roundtripBasepoint) {
+		t.Error("decode<>encode roundtrip produced different results")
+	}
+
+	encodedBasepoint := ristrettoBasepoint.Bytes()
+	if !bytes.Equal(compressedRistrettoBasepoint, encodedBasepoint) {
+		t.Error("point encode produced different results")
+	}
+}
+
+func TestRistrettoRandomRoundtrip(t *testing.T) {
+	// TODO quickcheck
+}
+
+func TestRistrettoSmallMultiplesTestVectors(t *testing.T) {
```

Comment:
Maybe add a reference to show that there are the original test vectors, https://ristretto.group/test_vectors/ristretto255.html ?